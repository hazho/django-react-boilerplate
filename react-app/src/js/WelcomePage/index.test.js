import React from "react";
import { render, screen } from "@testing-library/react";

import { WelcomePage } from "./index";
import { MemoryRouter } from "react-router-dom";

describe("Welcome", () => {
  it("renders without props", async () => {
    const component = render(
      <MemoryRouter>
        <WelcomePage />
      </MemoryRouter>
    );
    expect(component).toBeTruthy();
    expect(component.container.querySelector(".welcome__message")).toBeFalsy();
    expect(component.container).toMatchSnapshot();
  });

  it("calls fetchWelcomeMessage on mounting", async () => {
    const mockedFunc = jest.fn();
    expect(mockedFunc).not.toBeCalled();
    render(
      <MemoryRouter>
        <WelcomePage fetchWelcomeMessage={mockedFunc} />
      </MemoryRouter>
    );
    expect(mockedFunc).toBeCalled();
  });

  it("renders message if passed", async () => {
    const message = "Test message";
    render(
      <MemoryRouter>
        <WelcomePage message={message} />
      </MemoryRouter>
    );
    expect(screen.getByText(message)).toBeTruthy();
  });
});
