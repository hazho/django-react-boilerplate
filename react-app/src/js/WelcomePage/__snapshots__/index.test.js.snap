// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders without props 1`] = `
<div>
  <main
    class="welcome"
  >
    <h1
      class="welcome__title"
    >
      Welcome to our app!
    </h1>
    <p>
      Now, you may edit the routes on the \`routes.js\` file and/or edit this file to start developing your app :]
    </p>
    <p>
      To understand a bit better about routing, take a look at our
      <a
        href="/sample-nested-page/"
      >
         sample nested page
      </a>
      .
    </p>
  </main>
</div>
`;
